nikola/plugin_categories.py

```diff
@@ -273,7 +278,18 @@ def _read_extra_deps(self, post):
 
     def register_extra_dependencies(self, post):
         """Add dependency to post object to check .dep file."""
-        post.add_dependency(lambda: self._read_extra_deps(post), 'fragment')
+        def create_lambda(lang):
```

Comment:
Add a comment for future readers to explain this:

    # We create a lambda like this so we can pass `lang` to it, because if we didn’t add that function, `lang` would always be the last language in TRANSLATIONS.